hamilton/driver.py

```diff
@@ -1807,6 +1810,34 @@ def with_materializers(
         self.materializers.extend(materializers)
         return self
 
+    def with_cache(
+        self,
+        path: Optional[Union[str, pathlib.Path]] = None,
+        metadata_store: Optional[cache_store.BaseStore] = None,
+        result_store: Optional[cache_store.BaseStore] = None,
+        resume_from: Optional[Union[str, Literal["latest"]]] = None,
+    ) -> "Builder":
```

Comment:
```suggestion
    def with_cache(
        self,
        path: Optional[Union[str, pathlib.Path]] = None,
        metadata_store: Optional[cache_store.BaseStore] = None,
        result_store: Optional[cache_store.BaseStore] = None,
        **kwargs = None,
    ) -> "Builder":
```

Reply:
I think this API is fine -- kwargs can then handle strategy specific things, like `ignore`, etc.

Reply:
Decisions:

1. with_cache is ONLY for fingerprinting
2. remove **kwargs - and push configurable kwargs for the fingerprinting strategy.